tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5377,6 +5377,26 @@ class HLSLExternalSource : public ExternalSemaSource {
         return true;
       }
       QualType argType = arg.getAsType();
+      // Allow DependentType for case like
+      // template <typename T> void foo(GroupNodeInputRecords<T> data) {}
+      if (argType->isDependentType())
+        return false;
```

Comment:
This will need to be validated during instantiating right? Can you add tests for failure cases?

Reply:
Done.